skip `StubModule` mixins during package RBI generation

froydnj opened the pull request:
<!-- (optional) Explain your change, focusing on the details of the solution. This is a great place to call out user-visible changes. -->

I am not super-excited about this PR, but we can legitimately type-check codebases with RBIs containing slightly out-of-date definitions as far as their mixins go, so we should be able to generate package RBIs for such codebases too.

### Motivation
<!-- Why make this change? Describe the problem, not the solution. This can also be a link to an issue. -->

Robustness during package RBI generation.

### Test plan
<!-- If you did not write tests for this change, replace the message below explaining why not. Why we should be confident this change is correct? If you changed the website, please include a screenshot of the proposed changes. -->

See included automated tests.

froydnj commented:
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_LAZPekgS8SRC01
→ https://go/builds/bui_LAZPllJitCrWQo

jez commented:
Yeah I'm not super excited about this either. Is there any way we can look into this more closely?

froydnj commented:
> Yeah I'm not super excited about this either. Is there any way we can look into this more closely?

I agree that this is not ideal, but the behavior being proposed here seems consistent with other parts of Sorbet (e.g. class linearization, bits in LSP, CFG building, etc.).

I added a commit with a test that illustrates the problem, and updated the change to skip `StubModule` to also show how the PR fixes things.

jez commented on test/testdata/packager/rbi_gen/missing_methods.rbi:
```diff
@@ -0,0 +1,14 @@
+# typed: autogenerated
+
+# Simulates a situation where we've run missing methods to find things, but
+# we've updated the code to no longer define things referred to by the
+# generated RBIs.
+#
+# RBI generation should not output NonexistentThing, or StubModule.
```
this makes sense, thanks!
